fe/fe-core/src/main/java/com/starrocks/clone/TabletSchedCtx.java

```diff
@@ -705,6 +708,19 @@ private void reset(boolean reserveTablet) {
         }
     }
 
+    public void resetDecommissionReplicaState() {
+        if (decommissionedReplica != null
+                && decommissionedReplica.getState() == ReplicaState.DECOMMISSION
+                && decommissionedReplicaPreviousState != null) {
+            decommissionedReplica.setState(decommissionedReplicaPreviousState);
```

Comment:
Could we revert the state if the previous state is not a NORMAL state?

Reply:
Yes, and we must do it. If the previous state is clone, we should reset the state to clone so that it can be correctly scheduled. Replica in schema change state will never be choose to repair or balance.